shm/cargo.hpp

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include <string>
+
+class Cargo {
+public:
+    Cargo();
+    Cargo(std::string name, size_t amount, size_t basePrice);
+    
+    virtual ~Cargo() = default;
+
+    Cargo& operator+=(size_t amount);
+    Cargo& operator-=(size_t amount);
+    bool operator==(const Cargo& cargo) const;
```

Comment:
This should be =0, because Cargo should be interface. But I will accept this. 

```
W klasach Alcohol, Fruit, Item dopisz brakujące metody oraz ich implementacje.

// override from Cargo
size_t getPrice() const override;
std::string getName() const override { return name_; }
size_t getAmount() const override { return amount_; }
size_t getBasePrice() const override { return base_price_; }
Cargo& operator+=(size_t amount) override;
Cargo& operator-=(size_t amount) override;
bool operator==(Cargo& cargo) const override;
```